### PR TITLE
Fix the bug of incorrect model type identification

### LINKED ISSUE #810: 🐛 [Bug] Incorrect judgment of model type
##  Bug Description

When I provide a TorchScript model, Torch-TensorRT claims it is an nn.Module model

```
INFO: [Torch-TensorRT] - ir was set to default, using TorchScript as ir
INFO: [Torch-TensorRT] - Module was provided as a torch.nn.Module, trying to script the module with torch.jit.script. In the event of a failure please preconvert your module to TorchScript
```

## To Reproduce

Just run:

```
import torch
import torch.nn as nn
import torch_tensorrt as tt

class Model(nn.Module):
    def __init__(self):
        super(Model,self).__init__()
        self.conv=nn.Conv2d(3,3,3)

    def forward(self,x1):
        out=self.conv(x1)
        return out

a=Model().cuda().eval()
b=torch.jit.trace(a,torch.ones([1,3,20,20]).cuda())

compile_settings = {}
compile_settings["inputs"] = [tt.Input(shape = [1,3,20,20])]
tt.logging.set_reportable_log_level(tt.logging.Level.Graph)

tt.compile(b,**compile_settings)
```

## Expected behavior

Model type is correctly judged.

## Environment

> Build information about Torch-TensorRT can be found by turning on debug messages

 - Torch-TensorRT Version : 1.0.0
 - PyTorch Version: 1.10.0
 - CPU Architecture: Intel(R) Xeon(R) Platinum 8352Y CPU @ 2.20GHz
 - OS (e.g., Linux): CentOS 7
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): pip
 - Build command you used (if compiling from source):
 - Are you using local sources or building from archives:
 - Python version: 3.6.8
 - CUDA version: 11.4
 - GPU models and configuration: A30
 - Any other relevant information:

## Additional context

I have fixed this bug locally. After confirming the existence of this bug, I will open a PR to fix it.

### PR DESCRIPTION
**Pull request by cyfwry**
Signed-off-by: liangtao07 <liangtao07@meituan.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #810

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: py/torch_tensorrt/_compile.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def compile(module: Any, ir="default", inputs=[], enabled_precisions=set([_enums
     target_ir = _module_ir(module, ir)
     if target_ir == _IRType.ts:
         ts_mod = module
-        if isinstance(module, torch.nn.Module):
+        if not isinstance(module, torch.jit.ScriptModule):
```
Comment (narendasan) [inline py/torch_tensorrt/_compile.py:91]: This wont necessarily hold for future cases. For example say someone provides a module in FX format

Comment (cyfwry) [reply py/torch_tensorrt/_compile.py:91]: I would write a "_parse_module_type" method to solve this problem.

File: py/torch_tensorrt/_compile.py [hunk 2]
```diff
@@ -85,10 +103,10 @@ def compile(module: Any, ir="default", inputs=[], enabled_precisions=set([_enums
     Returns:
         torch.nn.Module: Compiled Module, when run it will execute via TensorRT
     """
-    target_ir = _module_ir(module, ir)
+    module_type, target_ir = _module_ir(module, ir)
```
[the remark this answers is not in the document]
Comment (cyfwry) [reply py/torch_tensorrt/_compile.py]: Done.

**Review by narendasan — COMMENTED** [review]
Cool!, feature wise this looks good. To merge 1. Can you run the linter on your code? 2. I think it would be nice to add the Module Type detection to the test suite so this error will get caught in the future.

**Comment by narendasan** [comment]
You can add the tests to test_api.py

**Review by carlear — APPROVED** [review]
Code conforms to Python style guidelines

**Comment by cyfwry** [comment]
> Cool!, feature wise this looks good. To merge 1. Can you run the linter on your code? 2. I think it would be nice to add the Module Type detection to the test suite so this error will get caught in the future.

1. I run `run_py_linter.py` locally without any problems.

**Comment by cyfwry** [comment]
> Cool!, feature wise this looks good. To merge 1. Can you run the linter on your code? 2. I think it would be nice to add the Module Type detection to the test suite so this error will get caught in the future.

For 2, done.

**Review by narendasan — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**